lua/impairative/operations.lua

```diff
@@ -51,25 +52,40 @@ local function process_desc(desc, i)
     end
 end
 
-local function maybe_with_better_n(args, callbacks)
+---@param args table operation options
+---@param default_better_n_args ImpairativeBetterNArgs default better-n arguments
+---@param callbacks { backward: fun(), forward: fun() } callbacks to be wrapped
+---@return { backward: fun(), forward: fun() }
+local function maybe_with_better_n(args, default_better_n_args, callbacks)
     if not args.better_n then
         return callbacks
     else
-        local wrapper = require('better-n').create {
+        local better_n_args = type(args.better_n) == 'boolean'
+            and default_better_n_args
+            or vim.tbl_extend('force', default_better_n_args, args.better_n)
+
+        local better_n = require 'better-n'
+        local forward_wrapper = better_n.create {
             previous = callbacks.backward,
             next = callbacks.forward,
         }
+        local backward_wrapper = better_n.create {
+            previous = callbacks.forward,
+            next = callbacks.backward,
+        }
         return {
-            backward = wrapper.previous,
-            forward = wrapper.next,
+            backward = better_n_args.relative_direction
+                and backward_wrapper.next
+                or forward_wrapper.previous,
+            forward = forward_wrapper.next,
```

Comment:
This seems... .wrong? Shouldn't `forward` depend on `better_n_args.relative_direction` too?

Also, I'd prefer if you'd use an `if` here, and just return one table in the `then` clause and another in the `else` clause. Makes the whole thing more readable.

Reply:
> This seems... .wrong? Shouldn't forward depend on better_n_args.relative_direction too?
No, `forward` does not depend on the setting, since `n` and `N` should be mapped to the forward and backward direction respectively.

> Also, I'd prefer if you'd use an if here, and just return one table in the then clause and another in the else clause. Makes the whole thing more readable.
Changed it.